app/helpers/submissions_helper.rb

```diff
@@ -106,13 +106,7 @@ def get_submissions_table_info(assignment, groupings)
           g[:tags] = grouping.tags
           g[:commit_date] = grouping.last_commit_date
           g[:has_files] = grouping.has_files_in_submission?
-          g[:late_commit] =
-            # TODO: Enable this check for Git backend. See issue #1866.
-            if MarkusConfigurator.markus_config_repository_type == 'git'
-              false
-            else
-              grouping.past_due_date?
-            end
+          g[:late_commit] = grouping.past_due_date?
```

Comment:
@adisandro The previous issue here was about performance, not correctness. Just want to make sure you're aware of that and tried this out on a fairly large set of submissions.

Reply:
Yes, I was very careful about performance when fixing git this summer, following that bug and the related discussion on the rugged bug tracker. The bottleneck was the function to detect if a file/directory changed with a specific commit, and I implemented all the optimizations suggested by the rugged people. Still, some repo operations are always going to be quite expensive in the worst case scenario, like having to walk all histories from top to the first commit.
I haven't tried this out with a lot of submissions though. My plan would be to enable it, try it in the real 207 scenario, and revert if there are problems.